packages/inngest/src/components/InngestFunction.ts

```diff
@@ -565,6 +567,29 @@ export namespace InngestFunction {
       finish?: TimeStr;
     };
 
+    /**
+     * Ensures that only one run of the function is active at a time for a given key.
+     * If a new run is triggered while another is still in progress with the same key,
+     * the new run will either be skipped or replace the active one, depending on the mode.
+     *
+     * This is useful for deduplication or enforcing exclusive execution.
+     */
+    singleton?: {
+      /**
+       * An optional key expression used to scope singleton execution.
+       * Each unique key has its own singleton lock. Event data can be referenced,
+       * e.g. "event.data.user_id".
+       */
+      key?: string;
+
+      /**
+       * Determines how to handle new runs when one is already active for the same key.
+       * - `"ignore"` skips the new run.
+       * - `"cancel"` cancels the existing run and starts the new one.
+       */
+      mode: "ignore" | "cancel";
```

Comment:
Just a thought: would `skip` be a clearer option than `ignore`? Since we have the "skipped" status elsewhere for skipped runs

Reply:
hmm good point i think it's more consistent